internal/app/referenceclient/impl.go

```diff
@@ -267,11 +278,6 @@ func (i *invoker) bidiStream(
 
 	var protoErr *v1.Error
 	for _, msg := range req.RequestMessages {
-		if err := ctx.Err(); err != nil {
-			// If an error was returned, convert it to a proto Error
-			protoErr = internal.ConvertErrorToProtoError(err)
-			break
-		}
```

Comment:
This is unnecessary. If the context is cancelled, it will get noticed and caught elsewhere. Also, we don't need to do anything special since at this point the client has cancelled the operation and won't be looking at any response.